### DISCUSSION TITLE
assignPriorityReturnRoute question

### DISCUSSION THREAD

**Discussion by apella12** [zwave-js/zwave-js-ui#3194] [General]
Todays project is to try to use zwave-js-ui to assign a priority return route via MQTT instead of using simplicity studio.  Trying to follow previous directions I came up with
This zwave/_CLIENTS/ZWAVE_GATEWAY-zwave-js-ui/api/assignPriorityReturnRoute/set', '{"args":[105,1,[],"100k"]}, but got this message in MQTT explorer
```
{
  "success": false,
  "message": "Unknown API",
  "args": [
    105,
    1,
    [],
    "100k"
  ],
  "origin": {
    "args": [
      105,
      1,
      [],
      "100k"
    ]
  }
```

**Comment by robertsLando** [comment]
I think you have to update your zui instance to latest. What's the version you are running?

**Comment by apella12** [comment]
zwave-js-ui: 8.18.1.6b838b5
zwave-js: 10.23.2
home id: 4056168225
home hex: 0xf1c43721

This will be the first upgrade since i downloaded for testing, so will need to figure that out.

**Reply by robertsLando** [reply]
> zwave-js-ui: 8.18.1.6b838b5

Return routes support has been added on 8.20 so you need to upgrade first

**Comment by apella12** [comment]
Thanks for the info.  The good news is I'm on 8.21.2c.  

The bad news is (and I do not think this is your area, but maybe you can redirect this) is I can't get access to the zstick anymore. This my first experience with snap (or any container-like app) and I had a heck of a time getting the zstick recognized initially. Of the over dozen things I tried, what I thought solved the problem initially (adding user "admin" to the dialout group) clearly was not.  I think it was/is some sort of permissions issue. 

The sequence of event was zwave-js-ui working, update (looks like a change from 195 to 210 in the files), no connection to zstick anymore.  I have collected a file of things I have tried and a picture of the UI device options for debugging. Will try anything else.
[snap whoami.txt](https://github.com/zwave-js/zwave-js-ui/files/12229995/snap.whoami.txt)
![ports](https://github.com/zwave-js/zwave-js-ui/assets/54947543/ef152cc7-5411-4a34-ac30-f33f05168c79)

**Reply by robertsLando** [reply]
@jmgiaever Is the snap master here, maybe he can help you with that

**Reply by jmgiaever** [reply]
Hi. Which OS are you using?

**Comment by apella12** [comment]
Release = Raspbian GNU/Linux 11 (bullseye)
   Kernel = Linux 6.1.21-v8+
  Platform = Raspberry Pi 4 Model B Rev 1.1

Don't know if it relevant to snap, but the other home automation system I have used with a different zwave binding requires an add-on or it can't access the serial port

edit: the frustrating thing is that I had this working, I just don't what I did

**Comment by jmgiaever** [comment]
Hi. Hmmm, do you have to use the serial-port plug? Have you tried to use only `raw-usb`? 

I know there's been a udev issue with the latest Debian. I don't know if that affect Radbian too. 

These links might help to check if the issue is in the OS:

https://www.reddit.com/r/debian/comments/1331wlr/comment/jkaq50z/?utm_source=reddit&utm_medium=web2x&context=3

https://github.com/systemd/systemd-stable/commit/451ba55fecd8b494add2001b3ca3c1915c8fd655

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1035094

**Comment by apella12** [comment]
The OS issues you noted appear to be related to symlinks to USB devices (they create a common name like myzstick and link it to the mfg data found in dmesg) so that linux doesn't change the name on a reboot and mess things up.   /dev/ttyUSB0 is the linux assigned name and in my simple system (only one USB device) has not been subject to change, so I haven't used them.

I may need more guidance on how to use `raw-usb`.  

Based on
```
openhabian@openhab:~ $ sudo snap interface raw-usb
name:    raw-usb
summary: allows raw access to all USB devices
plugs:
  - zwave-js-ui
slots:
  - snapd
```
I tried
`openhabian@openhab:~ $ sudo snap connect zwave-js-ui:raw-usb snapd`

but nothing seems to change (same before and after)?
```
openhabian@openhab:~ $ snap connections
Interface         Plug                          Slot               Notes
hardware-observe  zwave-js-ui:hardware-observe  :hardware-observe  -
network           zwave-js-ui:network           :network           -
network-bind      zwave-js-ui:network-bind      :network-bind      -
raw-usb           zwave-js-ui:raw-usb           :raw-usb           -
```
EDIT:  I thought I read in my journeys that some serial-port lines need to be in the snapcraft.yaml file.  There isn't any in mine
[snapcraft.yaml.txt](https://github.com/zwave-js/zwave-js-ui/files/12233354/snapcraft.yaml.txt)

Edit2:  I'm a noob on snap, but if zwave-js-ui was able to connect to my Zstick ttyUSB0 for a couple of weeks, it doesn't seem to me to be a Debian problem. I don't think a simple zwave-js-ui update should have changed anything in the Debian OS? OTOH I did find this post from a year ago about it seems there was a problem with a Rpi3 https://forum.snapcraft.io/t/acessing-serial-device-from-a-snap-package/29052.  At least to get me back to testing the zwave-js-ui application, is there a similar bluetooth or other workaround to get to /dev/ttyUSB0 (this is the ID of a 700 series Silabs stick) on a Rpi4 2G or did a "roll my own" get developed?

**Comment by apella12** [comment]
I abandoned the snap and regrouped to run zwave-js-ui by one of the other methods. The assignPriorityReturnGroup as formated in my orginal post works with the upgraded version
